src/server/actions/invite.actions.ts

```diff
@@ -0,0 +1,192 @@
+        }
+        logger.error({ message: 'Non-Clerk error during user creation', error, email })
+        throw new ActionFailure({ form: 'Could not create account due to an unexpected error.' })
+    }
+}, CreateAccountSchema)
+
+const ClaimInviteSchema = z.object({
+    inviteId: z.string(),
+})
+
+export const claimInviteAction = userAction(async ({ inviteId }) => {
+    const { user: authUser } = await actionContext()
+
+    const siUserId = authUser.id!
+    const { clerkId: clerkUserId } = await db
+        .selectFrom('user')
+        .select('clerkId')
+        .where('id', '=', siUserId)
+        .executeTakeFirstOrThrow()
+
+    const pendingUser = await db
+        .selectFrom('pendingUser')
+        .innerJoin('org', 'org.id', 'pendingUser.orgId')
+        .select([
+            'pendingUser.id as pendingUserId',
+            'pendingUser.email',
+            'pendingUser.orgId as localOrgDbId',
+            'pendingUser.isResearcher',
+            'pendingUser.isReviewer',
+            'org.slug as orgSlug',
+            'org.name as orgName',
+        ])
+        .where('pendingUser.id', '=', inviteId)
+        .where('pendingUser.claimedByUserId', 'is', null)
+        .executeTakeFirst()
+
+    if (!pendingUser) {
+        return { success: false, error: 'Invalid or already claimed invitation.' }
+    }
+
+    if (pendingUser.email !== authUser.primaryEmailAddress?.emailAddress) {
+        return { success: false, error: 'This invitation is for a different user. Please log out and try again.' }
+    }
+
+    const clerkOrg = await findClerkOrganization({
+        slug: pendingUser.orgSlug,
+    })
+
+    // 1. Update local DB
+    const orgMembershipDetails = await findOrCreateOrgMembership({
+        userId: siUserId,
+        slug: pendingUser.orgSlug,
+        isResearcher: pendingUser.isResearcher,
+        isReviewer: pendingUser.isReviewer,
+        isAdmin: false, // Default for invitees
+    })
+
+    try {
+        const client = await clerkClient()
+        // 2. Add to Clerk Organization (or update role if already member)
+        try {
+            await client.organizations.createOrganizationMembership({
+                organizationId: clerkOrg.id, // This MUST be the Clerk Organization ID
+                userId: clerkUserId,
+                role: orgMembershipDetails.isAdmin ? 'org:admin' : 'org:member',
+            })
+        } catch (e: unknown) {
+            if (isClerkApiError(e) && e.errors?.[0]?.code === 'duplicate_organization_membership') {
+                logger.info(
+                    `User ${clerkUserId} already member of Clerk org ${clerkOrg.id}. Role update might be needed separately if changed.`,
+                )
+            } else {
+                throw e // Re-throw if it's another error
+            }
+        }
+
+        // 3. Update Clerk User Public Metadata
+        const clerkUserToUpdate = await client.users.getUser(clerkUserId)
```

Comment:
this should be done when account is created, not after mfa

Reply:
done